megamek/src/megamek/client/ratgenerator/ForceDescriptor.java

```diff
@@ -0,0 +1,1696 @@
+	public static final int EXP_VETERAN = 2;
+	
+	public static final String[] ORDINALS = {
+		"First", "Second", "Third", "Fourth", "Fifth",
+		"Sixth", "Seventh", "Eighth", "Ninth", "Tenth"
+	};
+	
+	public static final String[] PHONETIC = {
+		"Alpha", "Bravo", "Charlie", "Delta", "Echo",
+		"Foxtrot", "Golf", "Hotel", "India", "Juliett",
+		"Kilo", "Lima", "Mike", "November", "Oscar",
+		"Papa", "Quebec", "Romeo", "Sierra", "Tango",
+		"Uniform", "Victor", "Whiskey", "X-ray", "Yankee",
+		"Zulu"
+	};
+	
+	public static final String[] GREEK = {
+		"Alpha", "Beta", "Gamma", "Delta", "Epsilon",
+		"Zeta", "Eta", "Theta", "Iota", "Kappa",
+		"Lambda", "Mu", "Nu", "Xi", "Omicron", "Pi",
+		"Rho", "Sigma", "Tau", "Upsilon", "Phi",
+		"Chi", "Psi", "Omega"
+	};
+	
+	public static final String[] LATIN = {
+		"Prima", "Secunda", "Tertia", "Quarta", "Quinta",
+		"Sexta", "Septima", "Octava", "Nona", "Decima"
+	};
+	
+	public static final String[] ROMAN = {
+	        "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX", "X",
+	        "XI", "XII", "XIII", "XIV", "XV", "XVI", "XVII", "XVIII", "XIX", "XX"
+	};
+	
+	private int index;
+	private String name;
+	private String faction;
+	private Integer year;
+	private Integer eschelon;
+	private int sizeMod;
+	private boolean augmented;
+	private Integer weightClass;
+	private Integer unitType;
+	private HashSet<EntityMovementMode> movementModes;
+	private HashSet<MissionRole> roles;
+	private String rating;
+	private Integer experience;
+	private Integer rankSystem;
+	private Integer coRank;
+	private HashSet<String> models;
+	private HashSet<String> chassis;
+	private HashSet<String> variants;
+	private CrewDescriptor co;
+	private CrewDescriptor xo;
+	private String camo;
+	
+	private HashSet<String> flags;
+	
+    private FormationType formationType;
+    private String generationRule;
+	private boolean topLevel;
+	private boolean element;
+	private int positionIndex;
+	private int nameIndex;
+	private String fluffName;
+	private Entity entity;
+	private List<ValueNode> nameNodes;
+	
+	private boolean generateAttachments;
+	private ForceDescriptor parent;
+	private ArrayList<ForceDescriptor> subforces;
+	private ArrayList<ForceDescriptor> attached;
+	private double dropshipPct = 0.0;
+	private double jumpshipPct = 0.0;
+	private double cargo = 0.0;
+	
+	public ForceDescriptor() {
+		faction = "IS";
+		year = 3067;
+		movementModes = new HashSet<EntityMovementMode>();
+		roles = new HashSet<MissionRole>();
+		formationType = null;
+		experience = EXP_REGULAR;
+		models = new HashSet<String>();
+		chassis = new HashSet<String>();
+		variants = new HashSet<String>();
+		parent = null;
+		subforces = new ArrayList<ForceDescriptor>();
+		attached = new ArrayList<ForceDescriptor>();
+		flags = new HashSet<String>();
+		topLevel = false;
+		element = false;
+		positionIndex = -1;
+		nameIndex = -1;
+		fluffName = null;
+	}
+	
+	public boolean matches(ChassisRecord cRec) {
```

Comment:
Let's add comments for these "matches" methods, as I'm not entirely sure what they do.